py/desispec/ccdcalib.py

```diff
@@ -423,28 +460,38 @@ def compute_nightly_bias(night, cameras, outdir=None, nzeros=25, minzeros=20,
         rank, size = 0, 1
 
     #- Find all zeros for the night
-    expids = None
+    expdict = None
     if rank == 0:
-        expids = _find_zeros(night)
+        expdict = _find_zeros(night,cameras=cameras,minzeros=nzeros)
 
     if comm is not None:
-        expids = comm.bcast(expids, root=0)
+        expdict = comm.bcast(expdict, root=0)
+
+    rawfiles_dict={}
+    skipped_cams=[]
+    for cam,expids in expdict.items():
+        if len(expids) < minzeros:
+            msg = f'Only {len(expids)} ZEROS on {night} and cam {cam}; need at least {minzeros}'
+            if rank == 0:
+                log.error(msg)
+            skipped_cams.append(cam)
+            continue
+
+        if len(expids) > nzeros:
+            nexps = len(expids)
+            n = (nexps - nzeros)//2
+            expdict[cam] = expids[n:n+nzeros]
 
-    if len(expids) < minzeros:
-        msg = f'Only {len(expids)} ZEROS on {night}; need at least {minzeros}'
         if rank == 0:
-            log.critical(msg)
-        raise RuntimeError(msg)
+            log.info(f'Using {len(expdict[cam])} ZEROs for nightly bias {night} and cam {cam}')
 
-    if len(expids) > nzeros:
-        nexps = len(expids)
-        n = (nexps - nzeros)//2
-        expids = expids[n:n+nzeros]
+        rawfiles_dict[cam] = [io.findfile('raw', night, e) for e in expids]
```

Comment:
You can move this logic to the next for loop so that each rank only determine the rawfiles for its camera(s)

Reply:
done